### DISCUSSION TITLE
Integration of heat maps into Remote Usabilty Lab

### DISCUSSION THREAD

**Discussion by uchiha-vivek** [ruxailab/RUXAILAB#354] [Ideas]
@KarinePistili 
Hi, My name is Vivek Sharma  and  have expericence of Full Stack Development.

I have few doubts regarding the above Project Idea.

Do you want to build a tool for heat mapping from scratch or the integration of third party feature like HotJar or Google Analytics to predict the heat map for the web applications. 
Note- The problems with these tools are that they are free for a limited amount of time and then they charge a monthly bill.

I have  experience of working with HotJar and Google Analytics in my previous company @Immer(SDE INTERN).
These tools give you good tracking of heat map and allows you track the behaviour of user .
You can see which part of the website is clicked most and which section is viewed most of the times . This gives you benefit to improve the
UI/UX of your website at the same time gives insights for the behavarioul  nature of the users.

Eye-tracking heat maps can be enhanced by putting attractive things on your web app like , mouse tracking can be enhanced in case  
the page is engaging and user wants to read more content and same goes for other features.

Apart from these there are other alternatives like Leaflet js and HeatMap js which can olny be used for Analytics of data and its visual representation.

Now which alternative you want as i am comfortable with both of these as i have prior experience in it,

Note- These tools can also be trained from scratch  using ML and NLP but then one has to use DOM Manipulations for web app like mouse event,click event ,scroll event .It would take a time  for extracting data but this can also be done. Here we have to also consider how much time user spent on the website and what are its particular interest to analyze the metrics and overall performance.

## CHALLENGES INCLUDED ##
Dynamic Url Tracking
Data analysis

Even using pre-built tools have some  issues like

1-Analysing the metrics of Data
2-Enhancement of Ui and proper positions of components.

I can work with all the above methods since i have experience of working with them but every method has some pros and cons asscociated with it. I would like mentors to discuss about it and  any  of their valuable suggestion would help me to make the project.Moreover i have ideas for some more features which we can add at top of our existing project.

**Comment by DarrenDsouza7273** [comment]
I don't think so they are aiming to integrate a third party tool they would rather create their own, keeping in mind the flexibility required as per their needs
